sympy/combinatorics/fp_groups.py

```diff
@@ -118,21 +118,46 @@ def identity(self):
     def __contains__(self, g):
         return g in self.free_group
 
-    def subgroup(self, gens, C=None):
+    def subgroup(self, gens, C=None, homomorphism=False):
         '''
         Return the subgroup generated by `gens` using the
         Reidemeister-Schreier algorithm
+        homomorphism -- When set to True, return a dictionary containing the images
+                     of the presentation generators in the original group.
+
+        Examples
+        ========
+        >>> from sympy import S
+        >>> from sympy.combinatorics.fp_groups import (FpGroup, low_index_subgroups,
+        ...                                    reidemeister_presentation, FpSubgroup)
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> F, x, y = free_group("x, y")
+        >>> f = FpGroup(F, [x**3, y**5, (x*y)**2])
+        >>> H = [x*y, x**-1*y**-1*x*y*x]
+        >>> K,T = f.subgroup(H, homomorphism=True)
+        >>> T(K.generators)
+        [x*y, x**-1*y**2*x**-1]
+        >>> T(K.generators[0]*K.generators[1])
+        x*y*x**-1*y**2*x**-1
+        >>> T(K.generators[0]*K.generators[1])
+        x*y*x**-1*y**2*x**-1
 
         '''
+
         if not all([isinstance(g, FreeGroupElement) for g in gens]):
             raise ValueError("Generators must be `FreeGroupElement`s")
         if not all([g.group == self.free_group for g in gens]):
                 raise ValueError("Given generators are not members of the group")
         g, rels = reidemeister_presentation(self, gens, C=C)
+        if homomorphism:
+            g, rels, _gens = reidemeister_presentation(self, gens, C=C, homomorphism=True)
```

Comment:
You end up running `reidemeister_presentation` twice if `homomorphism=True`, so better move the first one into an "else".